src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -40,64 +48,104 @@ def _market_stats_channel(self):
     def _grid_fees_channel(self):
         return f"{self.channel_prefix}/grid_fees"
 
-    def sub_to_area_event(self):
-        self.redis_db = self.area.config.external_redis_communicator
-        self.redis_db.sub_to_multiple_channels({
+    @staticmethod
+    def _get_transaction_id(payload):
+        data = json.loads(payload["data"])
+        if "transaction_id" in data and data["transaction_id"] is not None:
+            return data["transaction_id"]
+        else:
+            raise ValueError("transaction_id not in payload or None")
+
+    def _register(self, payload):
+        self._connected = register_area(self.redis_com, self.channel_prefix, self.connected,
+                                        self._get_transaction_id(payload),
+                                        area_uuid=self.area.uuid)
+
+    def _unregister(self, payload):
+        self._connected = unregister_area(self.redis_com, self.channel_prefix, self.connected,
+                                          self._get_transaction_id(payload))
+
+    def sub_to_external_channels(self):
+        self.redis_com = self.area.config.external_redis_communicator
+        sub_channel_dict = {
             f"{self.channel_prefix}/market_stats": self.market_stats_callback,
             f"{self.channel_prefix}/dso_market_stats": self.dso_market_stats_callback,
-            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback
-        })
+            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback,
+            f"{self.channel_prefix}/register_participant": self._register,
+            f"{self.channel_prefix}/unregister_participant": self._unregister}
+        if self.area.config.external_redis_communicator.is_enabled:
+            self.aggregator = self.area.config.external_redis_communicator.aggregator
```

Comment:
In order not to forget and to summarize our discussion about it, could you please add a TODO comment in the SimulationConfig class that we should find a more appropriate place for the external_redis_communicator object than the config class.

Reply:
Ah, yes, how about in the new added global_objects module in the frame of D3ASIM-2765:
https://github.com/gridsingularity/d3a/pull/879/files#diff-2756a77886549ce43c1ecfd27073b805 ?
Shall I add a story for that? I fear that A TODO will not change much soon ;-)

Reply:
Please go ahead to add a story, you are right that TODOs can be long-lived :) thanks!

Reply:
FYI: https://gridsingularity.atlassian.net/browse/D3ASIM-2868